crates/nix_rs/src/command.rs

```diff
@@ -121,30 +130,37 @@ impl NixCmd {
         T: std::str::FromStr,
         <T as std::str::FromStr>::Err: std::fmt::Display,
     {
-        let stdout = self.run_with_args_returning_stdout(args).await?;
+        let mut stdout_stream = self.run_with_args_returning_stdout_stream(args).await?;
+        let mut stdout = Vec::new();
+        stdout_stream
+            .read_to_end(&mut stdout)
+            .await
+            .map_err(CommandError::ChildProcessError)?;
         let v = &String::from_utf8_lossy(&stdout);
         let v = T::from_str(v.trim()).map_err(|e| FromStrError(e.to_string()))?;
         Ok(v)
     }
 
     /// Run nix with given args, returning stdout.
-    pub async fn run_with_args_returning_stdout(
+    pub async fn run_with_args_returning_stdout_stream(
         &self,
         args: &[&str],
-    ) -> Result<Vec<u8>, CommandError> {
+    ) -> Result<tokio::process::ChildStdout, CommandError> {
         let mut cmd = self.command();
         cmd.args(args);
+        cmd.stdout(Stdio::piped());
         trace_cmd(&cmd);
-        let out = cmd.output().await?;
-        if out.status.success() {
-            Ok(out.stdout)
-        } else {
-            let stderr = String::from_utf8(out.stderr)?;
-            Err(CommandError::ProcessFailed {
-                stderr: Some(stderr),
-                exit_code: out.status.code(),
-            })
-        }
+
+        let mut child = cmd.spawn()?;
+        let exit_status = child.wait().await?; // Wait for the child to finish
+
+        // FIXME: capture stderr. When I tried to do this, it would fail to
+        // stream stdout and block until the process finished.
```

Comment:
@shivaraj-bh So this PR is ready except for fixing this?

Reply:
Yes! I will have a look tomorrow at the FIXME

Reply:
Probably not, some tests seem to be failing. I will check that out too.

Reply:
I will move this to draft until I fix them both.